Plugin.pm

```diff
@@ -282,12 +282,58 @@ sub trackInfoMenu {
 	my $extid = $track->extid;
 	$extid ||= $url if $url =~ /^tidal:/;
 
-	return _objInfoMenu($client,
-		$extid,
-		$track->artistName || $remoteMeta->{artist},
-		$track->album ? $track->albumname : $remoteMeta->{album},
-		$track->title || $remoteMeta->{title},
-	);
+	my $artist = $track->remote ? $remoteMeta->{artist} : $track->artistName;
+	my $album  = $track->remote ? $remoteMeta->{album} : ( $track->album ? $track->album->name : undef );
+	my $title  = $track->remote ? $remoteMeta->{title} : $track->title;
+
+	my $search = cstring($client, 'SEARCH');
+	my $items = [];
+	push @$items, {
+		name => "$search " . cstring($client, 'ARTISTS') . " '$artist'",
+		type => 'link',
+		url => \&search,
+		image => 'html/images/artists.png',
+		passthrough => [ {
+			type => 'artists',
+			query => $artist,
+		} ],
+	} if $artist;
+
+	push @$items, {
+		name => "$search " . cstring($client, 'ALBUMS') . " '$album'",
+		type => 'link',
+		url => \&search,
+		image => 'html/images/albums.png',
+		passthrough => [ {
+			type => 'albums',
+			query => $album,
+		} ],
+	} if $album;
+
+	push @$items, {
+		name => "$search " . cstring($client, 'SONGS') . " '$title'",
+		type => 'link',
+		url => \&search,
+		image => 'html/images/playall.png',
+		passthrough => [ {
+			type => 'tracks',
+			query => $title,
+		} ],
+	} if $title;
+
+	my $tid = $track->remote ? $remoteMeta->{id} : undef;
+	push @$items, {
+		name => cstring($client, 'PLUGIN_TIDAL_TRACK_MIX'),
+		url => \&getTrackRadio,
+		image => 'plugins/TIDAL/html/mix_MTL_svg_stream.png',
```

Comment:
One more thing: make this item of type `playlist`. This should give you the playback controls on the menu item directly.

Reply:
👍